google/cloud/bigtable/examples/bigtable_examples_common.cc

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/bigtable/examples/bigtable_examples_common.h"
+#include "google/cloud/internal/getenv.h"
+
+namespace google {
+namespace cloud {
+namespace bigtable {
+namespace examples {
+
+Example::Example(std::map<std::string, CommandType> commands)
+    : commands_(std::move(commands)) {
+  std::string full_usage;
+  // Force each command to generate its Usage string, so we can provide a good
+  // usage string for the whole program. We need to create an Table
+  // object to do this, but that object is never used, it is passed to the
+  // commands, without any calls made to it.
+  for (auto const& kv : commands_) {
+    if (kv.first == "auto") continue;
+    try {
+      kv.second({});
+    } catch (Usage const& u) {
+      full_usage += "    ";
+      full_usage += u.what();
+      full_usage += "\n";
+    } catch (...) {
+      // ignore other exceptions.
+    }
+  }
+  full_usage_ = std::move(full_usage);
```

Comment:
Q. Why not just build within `full_usage_` in the first place?

Reply:
Done.